backend/corpora/lambdas/api/v1/collection.py

```diff
@@ -226,12 +195,60 @@ def check(key) -> bool:
     verify_collection_links(body, errors)
 
 
-@dbconnect
+def get_doi_link_node(body: dict, errors: list) -> Optional[dict]:
+    links = body.get("links", [])
+    dois = [link for link in links if link["link_type"] == ProjectLinkType.DOI.name]
+
+    if not dois:
+        return None
+
+    # Verify that a single DOI exists
+    if len(dois) > 1:
+        errors.append({"link_type": ProjectLinkType.DOI.name, "reason": "Can only specify a single DOI"})
+        return None
+
+    return dois[0]
+
+
+def curation_get_normalized_doi_url(doi: str, errors: list) -> Optional[str]:
+    # Regex below is adapted from https://bioregistry.io/registry/doi 'Pattern for CURIES'
+    curie_reference_regex = r"^\d{2}\.\d{4}.*$"
+    if not re.match(curie_reference_regex, doi):
+        errors.append({"name": ProjectLinkType.DOI.value, "reason": "DOI must be a CURIE reference."})
+        return None
+    return f"https://doi.org/{doi}"
+
+
+def corpora_get_normalized_doi_url(doi_node: dict, errors: list) -> Optional[str]:
```

Comment:
anything but corpora. How about portal?

Reply:
but it _is_ for corpora, no? Or did we already agree to no longer introduce that name anywhere...